### PR TITLE
Private tournament

### PR DESCRIPTION
**Pull request by freefal**

### CODE REVIEW

File: src/js/ui/tournament/detail/passwordForm.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    <form id="tournamentPasswordForm"
+    onsubmit={function(e: Event) {
+      e.preventDefault();
+      return join(e.target as HTMLFormElement);
+    }}>
+      <fieldset>
+        <div className={'select_input no_arrow_after' + (settings.tournament.private() ? '' : ' notVisible')}>
+          <div className="text_input_container">
+            <label>Password: </label>
+            <input type="text" id="tournamentPassword" className="passwordField" />
```
Comment (veloce) [inline src/js/ui/tournament/detail/passwordForm.tsx]: `<input type="password">` here

File: src/js/ui/tournament/newTournamentForm.tsx [hunk 2]
```diff
@@ -81,7 +90,10 @@ function create(form: HTMLFormElement) {
   const increment = (elements[4] as HTMLTextAreaElement).value;
   const duration = (elements[5] as HTMLTextAreaElement).value;
   const timeToStart = (elements[6] as HTMLTextAreaElement).value;
-  xhr.create(variant, position, mode, time, increment, duration, timeToStart)
+  const isPrivate = (elements[7] as HTMLInputElement).checked ? (elements[7] as HTMLInputElement).value : '';
```
Comment (veloce) [inline src/js/ui/tournament/newTournamentForm.tsx]: it's better to have a `boolean` here

File: src/js/ui/tournament/detail/passwordForm.tsx (hunk 1 above)
Comment (veloce) [inline src/js/ui/tournament/detail/passwordForm.tsx]: or maybe not actually... this is not like a login password.

**Comment by freefal** [comment]
Yeah, I took my cue from the website on this one, but don't feel strongly
one way or the other. Happy to have it be your call.

**Pull request merged**